tests/components/husqvarna_automower/test_lawn_mower.py

```diff
@@ -0,0 +1,169 @@
+"""Tests for lawn_mower module."""
+from copy import deepcopy
+from unittest.mock import AsyncMock, MagicMock, patch
+
+from aioautomower import AutomowerSession
+import pytest
+
+from homeassistant.components.husqvarna_automower import DOMAIN
+from homeassistant.components.husqvarna_automower.lawn_mower import (
+    AutomowerLawnMowerEntity,
+)
+from homeassistant.components.lawn_mower import LawnMowerActivity
+from homeassistant.config_entries import ConfigEntryState
+from homeassistant.core import HomeAssistant
+
+from .const import AUTOMOWER_CONFIG_DATA, AUTOMOWER_SM_SESSION_DATA, MWR_ONE_ID
+from .test_common import make_mower_data
+
+from tests.common import MockConfigEntry
+
+
+@pytest.mark.asyncio
+async def setup_entity(hass: HomeAssistant):
+    """Set up entity and config entry."""
+
+    config_entry = MockConfigEntry(
+        domain=DOMAIN,
+        data=AUTOMOWER_CONFIG_DATA,
+        entry_id="automower_test",
+        title="Automower Test",
+    )
+
+    config_entry.add_to_hass(hass)
+
+    deepcopy(AUTOMOWER_SM_SESSION_DATA)
+    mower = make_mower_data()
+
+    with patch(
+        "aioautomower.AutomowerSession",
+        return_value=AsyncMock(
+            name="AutomowerMockSession",
+            model=AutomowerSession,
+            data=mower,
+            register_data_callback=MagicMock(),
+            unregister_data_callback=MagicMock(),
+            connect=AsyncMock(),
+            action=AsyncMock(),
+        ),
+    ), patch(
+        "homeassistant.helpers.config_entry_oauth2_flow.async_get_config_entry_implementation",
+    ) as mock_impl:
+        await hass.config_entries.async_setup(config_entry.entry_id)
+        await hass.async_block_till_done()
+        assert config_entry.state == ConfigEntryState.LOADED
+        assert len(hass.config_entries.async_entries(DOMAIN)) == 1
+        mock_impl.assert_called_once()
+
+    return config_entry
+
+
+@pytest.mark.asyncio
+async def test_lawn_mower_state(hass: HomeAssistant) -> None:
+    """Test lawn_mower state."""
+    mower = make_mower_data()
+    await setup_entity(hass)
+    coordinator = hass.data[DOMAIN]["automower_test"]
+    lawn_mower = AutomowerLawnMowerEntity(mower.data[0], coordinator)
+
+    def set_state(state: str):
+        """Set new state."""
+        mower.data[0].attributes.mower.state = state
+
+    def set_activity(activity: str):
+        """Set new state."""
+        mower.data[0].attributes.mower.activity = activity
+
+    assert lawn_mower._attr_unique_id == f"{MWR_ONE_ID}_lawn_mower"
+    set_activity("")
```

Comment:
Why do we set it to empty string? I'd not use empty string to mean something in the library.